jaxpm/distributed.py

```diff
@@ -0,0 +1,151 @@
+from typing import Any, Callable, Hashable
+
+Specs = Any
+AxisName = Hashable
+
+try:
+    import jaxdecomp
+    distributed = True
+except ImportError:
+    print("jaxdecomp not installed. Distributed functions will not work.")
+    distributed = False
+
+from functools import partial
+
+import jax
+import jax.numpy as jnp
+from jax._src import mesh as mesh_lib
+from jax.experimental.shard_map import shard_map
+from jax.sharding import PartitionSpec as P
+
+# NOTE
+# This should not be used as a decorator
```

Comment:
why?

Reply:
Because the decorator is triggered at import time and the mesh is set as `empty` 
When the function is wrapped correctly, it will be rexecuted correctly every jit

Reply:
oh I see, thanks!